### PR TITLE
Assign-to-Share: Update email templates

### PR DESCRIPTION
**Pull request by thucdtran**
## Description
Update email templates for assign-to-share terminology
## Screenshots / Demo Video

Screenshot 2024-06-18 at 8 08 43 PM

## Testing
Added a new test with the SHARE_TERMINOLOGY_ENABLED environment variable set. Test covers that the new terminology is used.

### CODE REVIEW

File: packages/server/src/lib/email.js [hunk 1]
```diff
@@ -337,6 +353,50 @@ async function sendGrantAssignedEmailsForAgency(assignee_agency, grantDetail, as
     await asyncBatch(inputs, deliverEmail, 2);
 }
 
+async function sendGrantSharedEmailsForAgency(assignee_agency, grant, grantDetail, assignorUserId) {
```
Comment (jeffsmohan) [inline packages/server/src/lib/email.js:356]: One thing I've found a bit hard to follow in our codebase at times — and that I'm hoping we might be able to avoid introducing more of here — is when we partially update terminology in our code. (For example, is it "team" or "agency"? Are those synonyms or actually different concepts in our codebase?) 

It's totally natural to want to tweak user-facing language over time, but I'd love to see us try to maintain consistent naming conventions for a given concept in anything non-user-facing in our codebase. In general, that means when we make a user-facing terminology change like "assign" to "share", we should change the user-facing references but keep all internal code referencing "assign". (This assumes we're not interested in the effort of a full-scale naming change in the code, including things like database column renames, which is generally not worth it.)

Basically, I'd love to keep all internal code references as "assign", even as we're switching to calling it "share" with users. 

To put that into practice with this PR, that would mean naming this function `sendGrantAssignedEmailsForAgency` and renaming the existing function `sendGrantAssignedEmailsForAgencyDeprecated` or similar. (The second function should of course get cleaned up when we clean up the share terminology feature flag.)

Maybe I'm overly worried about such things, but I've personally been thrown off by this sort of thing in our codebase before. And these paper cuts tend to add up over time in a codebase. If you disagree and don't think we need to sweat this stuff, feel free to disregard — I'm just one volunteer in the codebase, not the arbiter of such things. :)

**Review by jeffsmohan — CHANGES_REQUESTED** [review]
Looking really good, and thanks for adding tests! Just a few pretty minor comments.

**Comment by ClaireValdivia** [comment]
Confirming from meeting discussion that we should keep "assign" language in share as Jeff suggested above

File: packages/server/__tests__/email/email.test.js [hunk 2]
```diff
@@ -47,6 +47,7 @@ describe('Email module', () => {
         process.env.NODEMAILER_PORT = NODEMAILER_PORT;
         process.env.NODEMAILER_EMAIL = NODEMAILER_EMAIL;
         process.env.NODEMAILER_EMAIL_PW = NODEMAILER_EMAIL_PW;
+        process.env.SHARE_TERMINOLOGY_ENABLED = false;
```
[the remark this answers is not in the document]
Comment (thucdtran) [reply packages/server/__tests__/email/email.test.js]: Done.

File: packages/server/src/lib/email.js (hunk 1 above)
Comment (thucdtran) [reply packages/server/src/lib/email.js]: Made the change!

File: packages/server/src/static/email_templates/_grant_assigned_body.html [hunk 3]
```diff
@@ -17,7 +17,7 @@
                                             <td align="left" style="padding:0;Margin:0">
                                                 <p
                                                     style="Margin:0;-webkit-text-size-adjust:none;-ms-text-size-adjust:none;mso-line-height-rule:exactly;font-family:arial, 'helvetica neue', helvetica, sans-serif;line-height:39px;color:#000000;font-size:26px">
-                                                    <strong>Grant Assigned</strong>
+                                                    <strong>{{grant_assigned_header}}</strong>
```
[the remark this answers is not in the document]
Comment (thucdtran) [reply packages/server/src/static/email_templates/_grant_assigned_body.html:20]: Pinged Claire here; I've since updated it to conditionally remove the assign terminology.

**Review by jeffsmohan — APPROVED** [review]
lgtm!
